src/zenml/artifacts/type_registry.py

```diff
@@ -44,7 +44,15 @@ def register_integration(
 
     def get_artifact_type(self, key: Type[Any]) -> List[Type["BaseArtifact"]]:
```

Comment:
at a glance, doesn't this pull request contain two functions that do more ore less the same thing? check a dictionary for the presence of a key, return the associated value to that if it exists and otherwise search for supertypes, returning that associated value only if there is exactly one, else raising an error? seems like that could be in a generic function to make sure if something changes in the future, it has to be changed in one place only.

Reply:
Good idea to implement a generic function. Right now, there is a hiccup in our implementation. If the key is not present but its supertypes are, we need to check whether the corresponding values are the same as well. In our `default_materializer_registry`, the values are actual `Materializer`s which we can compare between each other. However, in the `type_registry`, the values are `list`s so they are unhashable and incomparable. That's why the implementation differs just a little bit between them. I have an idea on how to fix this and will update this thread as it progresses.

Reply:
Updated the main thread!